src/zend/function.rs

```diff
@@ -36,3 +46,96 @@ impl FunctionEntry {
         Box::into_raw(Box::new(self))
     }
 }
+
+pub type Function = zend_function;
+
+impl Function {
+    pub fn try_from_function(name: &str) -> Option<Self> {
+        unsafe {
+            let res = zend_fetch_function_str(name.as_ptr() as *const i8, name.len());
+            if res.is_null() {
+                return None;
+            }
+            Some(*res)
+        }
+    }
+    pub fn try_from_method(class: &str, name: &str) -> Option<Self> {
+        match ClassEntry::try_find(class) {
+            None => None,
+            Some(ce) => unsafe {
+                let res = zend_hash_str_find_ptr_lc(
+                    &ce.function_table,
+                    name.as_ptr() as *const i8,
+                    name.len(),
+                ) as *mut zend_function;
+                if res.is_null() {
+                    return None;
+                }
+                Some(*res)
+            },
+        }
+    }
+
+    pub fn from_function(name: &str) -> Self {
+        match Self::try_from_function(name) {
+            Some(v) => v,
+            None => panic!("Expected function `{}` to exist!", name),
```

Comment:
Perhaps more of a architectural question, but is it better to just not offer methods that can panic? It seems to be correct, the user should always use `try_from_*` and call unwrap() if they're so sure it can't fail

Reply:
In my opinion, it makes sense to offer infallible variants of these two functions because there's a valid usecase for fetching native functions and native methods, which are always available, and forcing the use of try=>unwrap for known existing native functions seemed a bit unergonomic to me.
What do you think?

Reply:
Hmm yes I can see that, but is it always true that native methods / functions are always available? What about across different versions of PHP (as the caller, I don't want to worry about knowing that), or what about the use of functions in the INI `disabled_functions`.

I understand wanting to be ergonomic, but I value the trust in knowing I'm not going to get runtime errors to be higher. A rust panic in a PHP extension is also something I would _never_ want to happen especially!

Reply:
Hmmmm, understandable, I guess I can just directly export the function entries of common functions, fetching them on startup, instead.
Reverting that part for now then :)